### PR TITLE
[codex] canonicalize Tinker hyperparameter aliases

### PR DESCRIPTION
**Pull request by RaychevAngel**
## Summary
- canonicalize Manager/DecisionEngine legacy Tinker hyperparameter aliases (`epochs`, `max_seq_len`) into `num_epochs` / `max_seq_length` before AutoResearch proposal/search state sees them
- canonicalize alias patches returned by the proposer so the persisted config, diary state, and next Tinker run use supported V1 names
- keep unsupported Tinker knobs rejected after alias normalization

## Validation
- `python3 -m compileall src tests/test_autoresearch.py`
- `uv run --with pytest --with anthropic --with langgraph --with fastapi --with httpx python -m pytest tests/test_autoresearch.py -q`
- `uv run --with pytest --with anthropic --with langgraph --with datasets --with huggingface-hub --with requests python -m pytest tests/test_autoresearch.py tests/test_tinker_runner.py tests/test_e2e_propose.py -q`
- `uv run --with pytest --with anthropic --with langgraph --with fastapi --with httpx --with datasets --with huggingface-hub --with requests python -m pytest tests -q --ignore=tests/data_generator/test_mode_b_hf_retrieval.py`

## Notes
- I intentionally stopped a broad branch run without the `--ignore` after it entered the older ungated live Hugging Face retrieval test on this branch base. The dedicated no-live rerun above passed.
- This fixes the live API failure observed after the baseline Tinker run: `Unsupported Tinker SFT hyperparameter(s): ['epochs']`.

### CODE REVIEW

**Comment by RaychevAngel** [comment]
Superseded by consolidated replacement PR(s): #116. This old draft was part of the fragmented AutoResearch/budget stack.
